components/tournaments_list/filter_buttons.lua

```diff
@@ -0,0 +1,164 @@
+local Array = require('Module:Array')
+local Class = require('Module:Class')
+local String = require('Module:StringUtils')
+local Variables = require('Module:Variables')
+local Table = require('Module:Table')
+
+local FilterButtons = {}
+
+---@class Category
+---@field name string
+---@field query string?
+---@field items string[]?
+---@field defaultItems string[]?
+---@field transform function?
+---@field expandKey string?
+---@field expandable boolean?
+---@field order function?
+
+---@return Html
+function FilterButtons.getFromConfig()
+	return FilterButtons.get(require('Module:FilterButtons/Config').categories)
+end
+
+---Entrypoint building a set of FilterButtons
+---@param categories Category[]
+---@return Html
+function FilterButtons.get(categories)
+	Array.forEach(categories, FilterButtons._loadCategories)
+
+	local div = mw.html.create('div')
+
+	for index, category in ipairs(categories) do
+		Variables.varDefine('filterbuttons_defaults_' .. category.name, table.concat(category.defaultItems, ','))
+		local buttons = FilterButtons._getButtonRow(category)
+		if index > 1 then
+			buttons:css('margin-top', '-7px')
+		end
+		div:node(buttons)
+	end
+
+	return div
+end
+
+---@param category Category
+function FilterButtons._loadCategories(category)
+	if category.items then
+		if not category.defaultItems then
+			category.defaultItems = category.items
+		end
+	end
```

Comment:
```suggestion
	elseif Table.isNotEmpty(category.items) then
		if category.order then
 			Array.orderInPlaceBy(category.items, category.order)
 		end
 
		return category
	end
```

Reply:
alternatively could add
`---@field load fun(cat: FilterButtonCategory): FilterButtonCategory` to `FilterButtonCategory`
and then at the start of `FilterButtons._loadCategories(category)` do
```lua
function FilterButtons._loadCategories(category)
	if category.load then
		return category.load(category)
	end
```